File: functions/openssl/variants/go/3__booth_recode_w5_6.go
package main

func boothRecodeW5(in uint32) uint32 {
    s := ^(uint32((int32(in) >> 5) - 1))
    if s == 0 {
        return 0
    }
    d := 63 - in
    d = (d >> 1) + (d & 1)

    return (d << 1) + (s & 1)
}


func main(){
    boothRecodeW5(0)
}

File: functions/openssl/variants/c/3__booth_recode_w5_6.c
#include <stdint.h>

static unsigned int _booth_recode_w5(unsigned int in) 
{
    unsigned int s, d;

    s = ~((in >> 5) & 1U) + 1U;
    d = (~in & 63U) + 1U;
    d = (d & s) | (in & ~s);
    d = (d + (d & 1U)) >> 1U;

    return (d << 1U) | (s & 1U);
}


int main(){
    _booth_recode_w5(0);
}
